### ISSUE TITLE
Inventory random disappear

### ISSUE THREAD

**Issue by BeOrNot2Be** [BeOrNot2Be/CocktailBuilder#32]
App is designed on asynchronous redux storage. Async redux is a really tricky thing that has to have a lot of bug prevention mechanics. It was designed that way because requirements to client app (API endpoints and outputed data wasn't designed for that, so just to fetch something simple, sometimes it takes 4 calls and usage of local cache. ) The code come out kinda 'facinating' but if I would use regular redux or other storage, code would become complete legacy. One of the last versions actually had some prevention mechanisms for async storage and calls. But as I do see, the problem can actually go deep to redux storage design and async react components.DidMount() ( screen updates ). The only possible reason why it happens is the issues or really small delays between screen updates and storage data / actions delivery. 
Basically issue happens on opening app. The whole async design goes wrong because of delays and wrong boolean mark the that saying that data was updated, so app saves inventory that has nothing in it.
The best and easiest solution that I have for you is to just have if list < 1 then not save to server untill remove button action was submitted. So what we basically doing is that we are gonna submit zero in inventory only when user pressed a button ( before it gets ad actually updating / saving inventory on every page rerender )
Malware, internet, delays would produce pop-up that says that something happened with internet on our end and that we sorry for that and it's gonna be fixed soon
